### DISCUSSION TITLE
[opencv4 contrib] Ability to select specific module under opencv-contrib.

### DISCUSSION THREAD

**Discussion by kentf** [microsoft/vcpkg#26598] [General]
I'm testing the waters for the ability to be able to select specific modules under opencv-contrib with vcpkg.

Is it a desired feature?

For my use case, I need 1 module under contrib, namely `ximgproc`. All the other modules are not needed.

I'm using manifest mode with `vcpkg.json`, and I'm building for multiple platforms. Not all platforms are happy with all the features under contrib. For instance, ancient version of CentOS does not like openssl from vcpkg unless some specific set of packages are installed.

I could make an overlay of opencv4, but I'd rather not. I'd like to have it as a feature in the public and official port.

I've started prototyping, and at the moment I've added this to `opencv4/vcpkg.json` file.

```json
    "contrib-ximgproc": {
      "description": "opencv_contrib_ximgproc module",
      "dependencies": [

      ]
    },
```

In `opencv4/portfile.cmake`:

```cmake
...
elseif("contrib-ximgproc" IN_LIST FEATURES)

  list(APPEND ADDITIONAL_BUILD_FLAGS 
    "-DBUILD_opencv_ximgproc=ON"
    "-DBUILD_opencv_video=ON"
    "-DBUILD_opencv_dnn=ON"
    "-DBUILD_opencv_flann=ON"
    "-DBUILD_opencv_features2d=ON"
    "-DBUILD_opencv_calib3d=ON"
    )
endif()
```

Other modules would have similar names.

Any thoughts and or suggestions before I invest too much time into this?

**Comment by JackBoosY** [comment]
Yeah, you choice is right. And you may need to use overlay-port.

> For instance, ancient version of CentOS does not like openssl from vcpkg unless some specific set of packages are installed.

We are improving this.
